sdk/modelsrepository/azure-iot-modelsrepository/src/main/java/com/azure/iot/modelsrepository/ModelsRepositoryClientBuilder.java

```diff
@@ -0,0 +1,333 @@
+    }
+
+    /**
+     * Set the default dependency resolution option that the built client will use. This field will have a default value.
+     *
+     * @param modelDependencyResolution A DependencyResolutionOption value to force model resolution behavior.
+     * @return the updated ModelsRepositoryClientBuilder instance for fluent building.
+     */
+    public ModelsRepositoryClientBuilder modelDependencyResolution(ModelDependencyResolution modelDependencyResolution) {
+        this.modelDependencyResolution = modelDependencyResolution;
+        return this;
+    }
+
+    /**
+     * Set the service endpoint that the built client will communicate with. This field is mandatory to set.
+     *
+     * @param repositoryEndpoint Uri of the service in String format.
+     * @return the updated ModelsRepositoryClientBuilder instance for fluent building.
+     */
+    public ModelsRepositoryClientBuilder repositoryEndpoint(String repositoryEndpoint) {
+        DtmiConventions.convertToUri(repositoryEndpoint);
+        this.repositoryEndpoint = DtmiConventions.convertToUri(repositoryEndpoint);
+        return this;
+    }
+
+    /**
+     * Sets the {@link ModelsRepositoryServiceVersion} that is used when making API requests.
+     * <p>
+     * If a service version is not provided, the service version that will be used will be the latest known service
+     * version based on the version of the client library being used. If no service version is specified, updating to a
+     * newer version of the client library will have the result of potentially moving to a newer service version.
+     * <p>
+     * Targeting a specific service version may also mean that the service will return an error for newer APIs.
+     *
+     * @param serviceVersion The service API version to use.
+     * @return the updated ModelsRepositoryClientBuilder instance for fluent building.
+     */
+    public ModelsRepositoryClientBuilder serviceVersion(ModelsRepositoryServiceVersion serviceVersion) {
+        this.serviceVersion = serviceVersion;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpClient} to use for sending a receiving requests to and from the service.
+     *
+     * @param httpClient HttpClient to use for requests.
+     * @return the updated ModelsRepositoryClientBuilder instance for fluent building.
+     */
+    public ModelsRepositoryClientBuilder httpClient(HttpClient httpClient) {
+        this.httpClient = httpClient;
+        return this;
+    }
+
+    /**
+     * Sets the {@link HttpLogOptions} for service requests.
+     *
+     * @param logOptions The logging configuration to use when sending and receiving HTTP requests/responses.
+     * @return the updated ModelsRepositoryClientBuilder instance for fluent building.
+     * @throws NullPointerException If {@code httpLogOptions} is {@code null}.
+     */
+    public ModelsRepositoryClientBuilder httpLogOptions(HttpLogOptions logOptions) {
```

Comment:
I don't know what the standard is, but it feels like this method should be called `setHttpLogOptions()`

Reply:
I followed the pattern in all the other SDKs, with fluent classes, it's standard to just have the name of the property
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchClientBuilder.java